loader.js

```diff
@@ -86,47 +104,29 @@ async function processLayout(options, frontMatter, content, resourcePath, scans)
     '|'
   )})`
 
-  const extendedFm = await extendFrontMatter({
-    content,
-    frontMatter: {
-      ...frontMatter,
-      __resourcePath: resourcePath,
-      __scans: scans
-    },
-    phase: 'loader',
-    extendFm: pluginOpts.extendFrontMatter
-  })
 
   const matches = await new Promise((resolve, reject) => {
     glob(layoutMatcher, (err, matches) => err ? reject(err) : resolve(matches))
   })
 
   if (!matches.length) {
     throw new Error(
-      `File "${resourcePath}" specified "${frontMatter.layout}" as its layout, but no matching file was found at "${layoutMatcher}"`
+      `File "${resourcePath}" specified "${mergedFrontMatter.layout}" as its layout, but no matching file was found at "${layoutMatcher}"`
     )
   }
 
   const { onContent } = pluginOpts
   if (onContent && this._compiler.name === 'server') {
     onContent({
-      ...frontMatter,
-      ...extendedFm,
-      ...{ __resourcePath: resourcePath },
+      ...mergedFrontMatter,
       content
     })
   }
 
   // Import the layout, export the layout-wrapped content, pass front matter into layout
   return `import layout from '${normalizeToUnixPath(layoutPath)}'
 
-export * from '${normalizeToUnixPath(layoutPath)}'
```

Comment:
I went ahead and removed this line for now. If #37 gets reverted, then I'll rebase and it'll remove that from the diff. Otherwise it can just be reverted with this change until upstream is fixed.